### DISCUSSION TITLE
Provide time range, when GitHub's IP addresses updates

### DISCUSSION THREAD

**Discussion by UrbanovichA** [community/community#60011] [Actions]
### Select Topic Area

Question

### Body

Hi!
Im implemented solution for whitelist IP addresses for GitHubs Hosted Agents that are listed on [MetaAPI](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/about-githubs-ip-addresses) at Azure NSG. Those range updates weekly, its ok for us
and my question is:
At what time range, does this file updated?
Because i need to set time trigger for updating my NSG rules, that will run after file updated. Trigger is set for every day - once, but i need time range, to avoid build pipelines failed, if IP address range is not updated at NSG.

Thanks

**Comment by sohaibaumer** [comment]
The specific time range is not publicly documented.You can ensure your NSG rules stay up to date with GitHub's IP addresses by periodically checking for updates programmatically using the IP addresses file URL (https://api.github.com/meta). Alternatively, you can set up a webhook or event-based trigger to update your NSG rules whenever the file changes. Monitoring GitHub's status and announcements can also help you stay informed about any changes.
